### PR TITLE
test grandchild case as child module

### PR DESCRIPTION
**Pull request by snopoke**
@kaapstorm 

This is the use case that I mentioned yesterday that fails because of the 'None' in the `changed_ids_by_case_tag` dict.

### CODE REVIEW

**Comment by snopoke** [comment]
Build will fail.

File: corehq/apps/app_manager/tests/test_suite.py [hunk 1]
```diff
@@ -1032,6 +1032,44 @@ def _load_case(self, child_module_form, case_type, parent_module=None):
...
+        self.module_2.unique_id = 'm2'
+        self.app.new_form(2, 'grandchild form', None)
+
+        self.module_2.case_type = 'tadpole'
+        m2f0 = self.module_2.get_form(0)
+        self._load_case(m2f0, 'tadpole', parent_module=self.module_1)
+
+        self.module_2.root_module_id = self.module_1.unique_id
+
+        self.assertXmlPartialEqual(self.get_xml('child-module-grandchild-case'), self.app.create_suite(), "./entry")
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_suite.py]: E501 line too long (116 > 115 characters)

**Comment by kaapstorm** [comment]
Tracking in [FB 176836](http://manage.dimagi.com/default.asp?176836)

File: corehq/apps/app_manager/suite_xml.py [hunk 2]
```diff
@@ -1950,82 +1950,132 @@ def get_target_module(case_type, module_id, with_product_details=False):
...
+                try:
+                    # assume that all forms in the module have the same case management
+                    form = module_.get_form(0)
+                except FormNotFoundException:
+                    pass
+                else:
+                    if form.requires_case():
+                        datums_.extend(self.get_datum_meta_module(module_))
+                    datums_.extend(self.get_new_case_id_datums_meta(form))
+                    datums_.extend(self.get_extra_case_id_datums(form))
```
Comment (kaapstorm) [inline corehq/apps/app_manager/suite_xml.py]: Adding this line (commit f3aa4da) will hopefully simplify an awful merge conflict.

**Pull request merged**